vmware_guest - Add support for snapshot operations

Abdul-Anshad-A opened the pull request:
##### ISSUE TYPE
 - Feature Pull Request

##### COMPONENT NAME
vmware_guest

##### ANSIBLE VERSION
ansible 2.3.0 - devel

##### SUMMARY
The vmware_guest module lacks the ability to handle snapshot operations. This is an initial effort to support the same.

### Create snapshot
```
  - vmware_guest:
     hostname: 192.168.1.209
     username: administrator@vsphere.local
     password: vmware
     validate_certs: False
     name: dummy_vm
     snapshot_op:
         op_type: create
         name: snap1
         description: snap1_description
```

### Remove a snapshot
```
  - vmware_guest:
     hostname: 192.168.1.209
     username: administrator@vsphere.local
     password: vmware
     validate_certs: False
     name: dummy_vm
     snapshot_op:
         op_type: remove
         name: snap1
```

### Revert to a snapshot
```
  - vmware_guest:
     hostname: 192.168.1.209
     username: administrator@vsphere.local
     password: vmware
     validate_certs: False
     name: dummy_vm
     snapshot_op:
         op_type: revert
         name: snap1
```

### List all snapshots of a VM
```
  - vmware_guest:
     hostname: 192.168.1.209
     username: administrator@vsphere.local
     password: vmware
     validate_certs: False
     name: dummy_vm
     snapshot_op:
         op_type: list_all
```

### List current snapshot of a VM
```
  - vmware_guest:
     hostname: 192.168.1.209
     username: administrator@vsphere.local
     password: vmware
     validate_certs: False
     name: dummy_vm
     snapshot_op:
         op_type: list_current
```

### Remove all snapshots of a VM
```
  - vmware_guest:
     hostname: 192.168.1.209
     username: administrator@vsphere.local
     password: vmware
     validate_certs: False
     name: dummy_vm
     snapshot_op:
         op_type: remove_all
```

gregdek commented:
Thanks @Abdul-Anshad-A. To the current maintainers, @jctanner, @jcpowermac, @mtnbikenc, @chrrrles please review according to guidelines (http://docs.ansible.com/ansible/developing_modules.html#module-checklist) and comment with text 'shipit', 'needs_revision' or 'close_me' as appropriate.

[This message brought to you by your friendly Ansibull-bot.]

jctanner commented:
@dav1x can you take a look and give your thoughts?

Abdul-Anshad-A commented on cloud/vmware/vmware_guest.py:
```diff
@@ -868,6 +933,104 @@ def run_command_in_guest(self, vm, username, password, program_path, program_arg
+        try:
+            snapshot_op_name = snapshot_op['op_type']
+        except KeyError:
+            self.module.fail_json(msg="specify name & description to create snapshot")
```
My bad, msg should be ""Specify op_type - create/remove/revert/list_all/list_current/remove_all"

dav1x commented:
@jctanner ack! 

@Abdul-Anshad-A 
Tested and all functionality is there. I tried to create a snapshot without a description. If you manually create a snapshot the only required field is the name. Should description be required?

Otherwise LGTM

gregdek commented:
Thanks @Abdul-Anshad-A for this PR. This PR requires revisions, either because it fails to build or by reviewer request. Please make the suggested revisions. When you are done, please comment with text 'ready_for_review' and we will put this PR back into review.

[This message brought to you by your friendly Ansibull-bot.]

Abdul-Anshad-A commented:
@jctanner @dav1x 
Did the following changes:
- Make description optional during snapshot create.
- Corrected display msg when op_type is not specified
- Added description for snapshot_op

Abdul-Anshad-A commented:
ready_for_review

gregdek commented:
Thanks @Abdul-Anshad-A. To the current maintainers, @jctanner, @jcpowermac, @mtnbikenc, @chrrrles please review according to guidelines (http://docs.ansible.com/ansible/developing_modules.html#module-checklist) and comment with text 'shipit', 'needs_revision' or 'close_me' as appropriate.

[This message brought to you by your friendly Ansibull-bot.]
